ethereum/contracts/pyth/PythGovernanceInstructions.sol

```diff
@@ -0,0 +1,156 @@
+// contracts/GovernanceStructs.sol
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.0;
+
+import "../libraries/external/BytesLib.sol";
+import "./PythInternalStructs.sol";
+
+/**
+ * @dev `PythGovernanceInstructions` defines a set of structs and parsing functions
+ * for Pyth governance instructions.
+ */
+contract PythGovernanceInstructions {
+
+    using BytesLib for bytes;
+
+    // Magic is `PTGM` encoded as a 4 byte data: Pyth Governance Message
+    uint32 constant MAGIC = 0x5054474d;
+
+    enum GovernanceModule {
+        Core, // 0
+        Target, // 1
+        Attest // 2
+    }
+
+    GovernanceModule constant MODULE = GovernanceModule.Target;
+
+    enum GovernanceAction {
+        UpgradeContract, // 0
+        SetGovernanceDataSource, // 1
+        SetDataSources, // 2
+        SetFee, // 3
+        SetValidPeriod // 4
+    }
+
+    struct GovernanceInstruction {
+        GovernanceModule module;
+        GovernanceAction action;
+        uint16 targetChainId;
+        bytes payload;
+    }
+
+    struct UpgradeContractPayload {
+        address newImplementation;
+    }
+
+    struct SetGovernanceDataSourcePayload {
+        PythInternalStructs.DataSource newGovernanceDataSource;
+        uint64 initialSequence;
+    }
+
+    struct SetDataSourcesPayload {
+        PythInternalStructs.DataSource[] dataSources;
+    }
+
+    struct SetFeePayload {
+        uint newFee;
+    }
+
+    struct SetValidPeriodPayload {
+        uint newValidPeriod;
+    }
+
+    /// @dev Parse a GovernanceInstruction
+    function parseGovernanceInstruction(bytes memory encodedInstruction) public pure returns (GovernanceInstruction memory gi) {
+        uint index = 0;
+
```

Comment:
Are you checking the magic number here?

Reply:
Ah right, I forgot to add magic here totally. Will do. Thanks.